cmd/kpod/spec.go

```diff
@@ -366,16 +371,43 @@ func (c *createConfig) GetVolumeMounts() []spec.Mount {
 		if len(spliti) > 2 {
 			options = strings.Split(spliti[2], ",")
 		}
-		// always add rbind bc mount ignores the bind filesystem when mounting
 		options = append(options, "rbind")
+		var foundrw, foundro, foundz, foundZ bool
+		for _, opt := range options {
+			switch opt {
+			case "rw":
+				foundrw = true
+			case "ro":
+				foundro = true
+			case "z":
+				foundz = true
+			case "Z":
+				foundZ = true
+			}
+		}
+		if !foundro && !foundrw {
+			// rw option is default
+			options = append(options, "rw")
```

Comment:
This looks wrong.  It should just say 
if foundro
 +			options = append(options, "ro")

Reply:
no this is fine. If the user doesn't give "rw" or "ro" as an option we wan't to default to "rw", otherwise the user has specified one of those as an option already.

Reply:
Where does the options = append(options, "ro")  call happen?

Reply:
If you look at line 372, the options are stored in the `options` array after being split. So if the user does `/var/lib/foo:/run/foo:Z,ro`, options will have the "ro" in it. But if the user just does `/var/lib/foo:/run/foo:Z`, foundro and foundrw will be false, so "rw" will be added to the options array.

Reply:
I don't think it hurts, but you probably do not need to do this. since rw is the default for mount unless user specifies ro.